From 3cccaf99b2cbaf21af45021b4578cb02460bc0f2 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 20 Mar 2015 14:44:30 +0800
Subject: [PATCH] progress.py: limit status updating frequency

While installing package very quickly, the setSubTopic and
setSub was invoked very frequently, and both of them reset
self.__lasttime which caused the frequency control not work

Upstream-Status: inappropriate [wr-installer requires]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 smart/progress.py | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/smart/progress.py b/smart/progress.py
index f98cf02..67c0df0 100644
--- a/smart/progress.py
+++ b/smart/progress.py
@@ -23,7 +23,7 @@ import thread
 import time
 import sys
 
-INTERVAL = 0.1
+INTERVAL = 1.0
 
 class Progress(object):
 
@@ -175,8 +175,8 @@ class Progress(object):
     def setSubTopic(self, subkey, subtopic):
         self.__lock.acquire()
         try:
-            if subkey not in self.__subtopic:
-                self.__lasttime = 0
+            #if subkey not in self.__subtopic:
+            #    self.__lasttime = 0
             self.__subtopic[subkey] = subtopic
         finally:
             self.__lock.release()
@@ -194,11 +194,11 @@ class Progress(object):
                 return
             if subkey not in self.__subtopic:
                 self.__subtopic[subkey] = ""
-                self.__lasttime = 0
+                #self.__lasttime = 0
             if subcurrent > subtotal:
                 subcurrent = subtotal
-            if subcurrent == subtotal:
-                self.__lasttime = 0
+            #if subcurrent == subtotal:
+            #    self.__lasttime = 0
             self.__subprogress[subkey] = (subcurrent, subtotal,
                                           fragment, subdata)
         finally:
-- 
1.9.1

